test/node/bigint.test.ts

```diff
@@ -0,0 +1,111 @@
+import { BSON } from '../register-bson';
+import { bufferFromHexArray } from './tools/utils';
+import { expect } from 'chai';
+
+describe('BSON BigInt deserialization support', function () {
+  describe('BSON.deserialize()', function () {
+    let testSerializedDoc: Buffer;
+    before(function () {
+      testSerializedDoc = bufferFromHexArray(['12', '6100', '2300000000000000']);
+    });
+
+    it('deserializes int64 to Number when useBigInt64,promoteValues, promoteLongs are default', function () {
```

Comment:
This is a perfect opportunity for making table tests, we want to make sure we cover the cases where all three variables are set to true/false/undefined (unset) so we can generate that along with the expected outcome. I'll slack you some examples.